src/library/scala/concurrent/impl/Promise.scala

```diff
@@ -377,8 +377,9 @@ private[concurrent] object Promise {
     private[this] final var _ec: ExecutionContext,
     private[this] final var _arg: Try[F],
     private[this] final val _xform: Int
-  ) extends DefaultPromise[T]() with Callbacks[F] with Runnable with Batchable with OnCompleteRunnable {
-    final def this(xform: Int, f: _ => _, ec: ExecutionContext) = this(f.asInstanceOf[Any => Any], ec.prepare(), null, xform)
+  ) extends DefaultPromise[T]() with Callbacks[F] with Runnable with Batchable {
+    final def this(xform: Int, f: _ => _, ec: ExecutionContext) =
```

Comment:
https://github.com/akka/akka/issues/29980 🤔

Reply:
I'm concerned about how closely this is holding the standard library to implementation details

Reply:
It's not, IMO.  It's on Akka for relying on it.